### PR TITLE
[CloudPC]Add 3 new onprem connection error codes (516,517,518)

### PR DESCRIPTION
**Pull request by wcchen101**
Add new Intune error types

### CODE REVIEW

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
@wcchen101 

1.Please update the `Microsoft.CloudManagedDesktop.json` changelog. It should include all your changes:
image

2.Please create a roadmap workitem and then attach the workitem number in this PR. Here is an example: https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/12472/

If you don't have access, please do this firstly to gain access:

Request access via [Improved ADO Access - 1ES Wikirequest to join](https://www.1eswiki.com/wiki/Improved_ADO_Access) .
Join the Microsoft Graph Roadmap Users ([graphroadmap@microsoft.com](mailto:graphroadmap@microsoft.com)) security group on [idweb](https://idwebelements/GroupManagement.aspx?Group=graphroadmap&Operation=join) .

**Review by AshleyYangSZ — APPROVED** [review]

**Comment by RuiHou105** [comment]
@wcchen101  please create roadmap workitem and attach number in this PR. And then I will approve this PR :)

**Comment by wcchen101** [comment]
Roadmap work item Id: 14553

**Review by RuiHou105 — APPROVED** [review]

**Comment by RuiHou105** [comment]
@Lauragra , @JarbasHorst , could you please help review it as the graph schema already went alive. Thanks!

File: api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md [hunk 1]
```diff
@@ -80,6 +80,9 @@ The result of a cloud PC Azure network connection health check.
 |resourceAvailabilityCheckResourceGroupLockedForReadonly|The selected resource group is locked and can't be modified for provisioning. Please remove this lock to allow provisioning to succeed.|
 |resourceAvailabilityCheckResourceGroupLockedForDelete|The selected resource group or its parent scope has been locked for delete actions. It may be because the IP addresses are used up. Please remove the lock and try again.|
 |resourceAvailabilityCheckTransientServiceError|The resource availability check failed due to a transient error. Please try it again. If the issue persists, please contact customer support.|
+|resourceAvailabilityCheckNoIntuneReaderRoleError|Do not have sufficient Intune reader role to check Intune platform restriction settings.|
```
Comment (JarbasHorst) [inline api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md:83]: What does not have sufficient Intune reader role? The Cloud PC? If I understand this error correctly, I think that we can change this description to be:

- "The (subject??? - Cloud PC???) doesn't have sufficient permissions to check the Intune platform restriction settings. Please make sure that the (subject??? - Cloud PC???) has the *Reader (What reader permission exactly?? Cloud PC Reader??)* permissions on Intune.

**Comment by JarbasHorst** [comment]
Hello @wcchen101, I reviewed your PR and left one comment. Please, take a look when you will have time.

Also, to avoid merge conflicts, please merge PR [17662](https://github.com/microsoftgraph/microsoft-graph-docs/pull/17662) into your branch to include my fixes and before addressing my comment.

File: api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md (hunk 1 above)
Comment (wcchen101) [reply api-reference/beta/resources/cloudpconpremisesconnectionhealthcheck.md:83]: Sure. Updated. Please check it again. Thanks.

**Comment by JarbasHorst** [comment]
@Lauragra this PR is ready for final review. Please, merge PR [17778](https://github.com/microsoftgraph/microsoft-graph-docs/pull/17778) into this branch to include my final edits.

**Comment by RuiHou105** [comment]
Hi @Lauragra , I see all comments are resolved and you also helped merge @JarbasHorst 's fix into this PR. So could you please help approve this PR as the graph schema already went alive? Thanks :)

**Review by Lauragra — APPROVED** [review]

**Pull request merged**